monai/bundle/scripts.py

```diff
@@ -116,6 +121,216 @@ def _get_fake_spatial_shape(shape: Sequence[Union[str, int]], p: int = 1, n: int
+        bundle_dir: target directory to store the download data.
+            Default is `bundle` subfolder under`torch.hub get_dir()`.
+        source: the place that saved the bundle.
+            If `source` is `github`, the bundle should be within the releases.
+        repo: the repo name. If `None` and `url` is `None`, it must be provided in `args_file`.
+            If `source` is `github`, it should be in the form of `repo_owner/repo_name`.
+            For example: `Project-MONAI/MONAI`.
+        url: the url to download the data. If not `None`, data will be downloaded directly
+            and `source` will not be checked.
+            If `name` contains the filename, it will be used as the downloaded filename (without postfix).
+            Otherwise, the filename is determined by `monai.apps.utils._basename(url)`.
+        progress: whether to display a progress bar.
+        args_file: a JSON or YAML file to provide default values for all the args in this function.
+            so that the command line inputs can be simplified.
+
+    """
+    _args = _update_args(
+        args=args_file, name=name, bundle_dir=bundle_dir, source=source, repo=repo, url=url, progress=progress
+    )
+
+    if "name" not in _args and url is None:
+        raise ValueError(f"To download from source: {source}, `name` must be provided.")
+    if "repo" not in _args and url is None:
+        raise ValueError(f"To download from source: {source}, `repo` must be provided.")
+    _log_input_summary(tag="download", args=_args)
+    name_, bundle_dir_, source_, repo_, url_, progress_ = _pop_args(
+        _args, name=None, bundle_dir=None, source="github", repo=None, url=None, progress=True
+    )
+
+    if bundle_dir_ is None:
+        get_dir, has_home = optional_import("torch.hub", name="get_dir")
+        if has_home:
+            bundle_dir_ = Path(get_dir()) / "bundle"
+        else:
+            raise ValueError("bundle_dir=None, but no suitable default directory computed. Upgrade Pytorch to 1.6+ ?")
+    bundle_dir_ = Path(bundle_dir_)
+
+    filename: Optional[str] = None
+    if name_ is not None and len(name_.split(ID_SEP_KEY)) == 2:
+        name_, filename = name_.split(ID_SEP_KEY)
+
+    if url_ is not None:
+        if filename is not None:
+            filepath = bundle_dir_ / f"{filename}"
+        else:
+            filepath = bundle_dir_ / f"{_basename(url_)}"
+        download_url(url=url_, filepath=filepath, hash_val=None, progress=progress_)
+        filepath = Path(filepath)
+        if filepath.name.endswith(("zip", "tar", "tar.gz")):
+            extractall(filepath=filepath, output_dir=bundle_dir_, has_base=True)
+    elif source_ == "github":
+        _download_from_github(
+            repo=repo_, tag_name=name_, download_path=bundle_dir_, filename=filename, progress=progress_
+        )
+    else:
+        raise NotImplementedError("So far, only support to download from url or `github` source.")
```

Comment:
"Currently only download from provided URL in (`url` argument) or Github (`source` == "github") is implemented" might be a more informative message.